rudder-core/src/main/scala/com/normation/rudder/db/Doobie.scala

```diff
@@ -123,6 +122,16 @@ object Doobie {
       case \/-(a) => Full(a)
     }
   }
+  implicit def xorBoxToBox[A](res: \/[Throwable, Box[A]]): Box[A] = res match {
+    case -\/(e) => Failure(e.getMessage, Full(e), Empty)
+    case \/-(a) => a
+  }
+  implicit class XorBoxToBox[A](res: \/[Throwable, Box[A]]) {
+    def box = res match {
+      case -\/(e) => Failure(e.getMessage, Full(e), Empty)
+      case \/-(a) => a
+    }
```

Comment:
both are surprinsignly similar - is there a reason to have both ?

Reply:
box method is never used, are you sure you need to define the implicit class ?

If so, Iwould do "def box = xorBoxToBox(res)"  to prevent code duplication

Reply:
Both are similar but there is a Box more in param for the second. I need both, and I could express one with the other at the price of a class instance+deconstruction more (for nothing). I can express classes from methods.